src/PostgREST/Query/Statements.hs

```diff
@@ -156,8 +156,7 @@ prepareCall returnsScalar returnsSingle callProcQuery selectQuery countQuery cou
      | getMediaType mt == MTGeoJSON                       = asGeoJsonF
      | isJust binaryField && getMediaType mt == MTTextXML = asXmlF $ fromJust binaryField
      | isJust binaryField                                 = asBinaryF $ fromJust binaryField
-     | returnsSingle && not multObjects                   = asJsonSingleF returnsScalar
-     | otherwise                                          = asJsonF returnsScalar
+     | otherwise                                          = asJsonF returnsScalar returnsSetOfScalar returnsSingleComposite
```

Comment:
This is really the change I'm after. This way the `Statements` module only wraps the queries and decides on an aggregation to apply. This makes https://github.com/PostgREST/postgrest/pull/1582 easier to implement(for cases where the json aggregation is overridden).